progression/src/main/java/me/mykindos/betterpvp/progression/profession/woodcutting/specialitems/RebirthOfMagnoliaWeapon.java

```diff
@@ -0,0 +1,135 @@
+        blockStates.stream()
+                .filter(blockState -> blockState.getType().equals(Material.TALL_GRASS))
+
+                // Tall grass is like a 2-block block, so we only want the bottom one
+                // You need to get the block up 1 because event.getBlock() returns the block that was bonemealed
+                .filter(blockState -> blockState.getY() == event.getBlock().getRelative(0, 1, 0).getY())
+                .forEach(blockState -> blockState.setType(lootTypes.random()));
+
+        // Remove any leftover tall grass
+        blockStates.removeIf(blockState -> blockState.getType().equals(Material.TALL_GRASS));
+    }
+
+    // TODO: MOVE TO UTIL BLOCK EVENTUALLY WHEN ALL OF WOODCUTTING IS REFACTORED
+    public boolean isFlower(BlockState blockState) {
+        // List of all flowers in Minecraft (can expand this list as needed)
+        return switch (blockState.getType()) {
+            case DANDELION, POPPY, BLUE_ORCHID, ALLIUM, AZURE_BLUET,
+                    RED_TULIP, ORANGE_TULIP, WHITE_TULIP, PINK_TULIP,
+                    OXEYE_DAISY, CORNFLOWER, LILY_OF_THE_VALLEY, WITHER_ROSE ->
+                    true;
+            default -> false;
+        };
+    }
+    @Override
+    public List<Component> getLore(ItemMeta meta) {
+        List<Component> lore = new ArrayList<>();
+        lore.add(UtilMessage.deserialize("Long ago, the Magnolia Sovereign wept for a"));
+        lore.add(UtilMessage.deserialize("fading world. Her tears blessed barren soil"));
+        lore.add(UtilMessage.deserialize("with life, giving rise to extraordinary flowers."));
+        lore.add(UtilMessage.deserialize("This essence carries her power, blooming <orange>rare</orange>"));
+        lore.add(UtilMessage.deserialize("<orange>flowers</orange> wherever it touches the ground."));
+        return lore;
+    }
+
+    @Override
+    public void loadWeaponConfig() {
+        lootTypes = new WeighedList<>();
+
+        ConfigurationSection lootSection = getConfigObject("loot", null, ConfigurationSection.class);
+
+        // Probably should create a section here
+        if (lootSection == null) {
+            log.error("Failed to load flower loot types").submit();
+            return;
+        }
+
+        for (String lootItemKey : lootSection.getKeys(false)) {
+            ConfigurationSection lootItemSection = lootSection.getConfigurationSection(lootItemKey);
+            if (lootItemSection == null) continue;
+
+            String itemMaterialAsString = lootItemSection.getString("material");
+            if (itemMaterialAsString == null) continue;
+
+            Material material = Material.getMaterial(itemMaterialAsString.toUpperCase());
+            if (material == null) continue;
+
+            int frequency = lootItemSection.getInt("frequency");
+
+            lootTypes.add(frequency, 1, material);
+        }
+
+        log.info("Loaded " + lootTypes.size() + " flower loot types").submit();
```

Comment:
UtilItem.getDropTables is probably better, keeps it consistent with stuff going forward